uk.co.compucorp.civicrm.hrcore/tests/phpunit/CRM/HRCore/Service/CiviHRStatsGathererTest.php

```diff
@@ -1,35 +1,296 @@
+    $this->assertEquals(1, $stats->getContactSubtypeCount('CatClub'));
+    $this->assertEquals(2, $stats->getContactSubtypeCount('Cat'));
+    $this->assertEquals($expectedOrgs, $organizationCount);
+    $this->assertEquals($expectedIndividuals, $individualCount);
+    $this->assertEquals($expectedContacts, $stats->getEntityCount('contact'));
+  }
+
+  public function testDeletedEntitiesWillNotBeIncluded() {
+    $existingContactCount = $this->getCurrentContactCount();
+    $deletedContact = ContactFabricator::fabricate();
+    $deletedContactID = $deletedContact['id'];
+    civicrm_api3('Contact', 'delete', ['id' => $deletedContactID]);
+    $params = ['is_deleted' => 1, 'id' => $deletedContactID];
+    $stillExists = civicrm_api3('Contact', 'getcount', $params) == 1;
+
+    if (!$stillExists) {
+      $this->markTestSkipped('There was an error in soft deletion of contact');
+    }
+
+    $stats = $this->getGatherer()->gather();
+
+    $this->assertEquals($existingContactCount, $stats->getEntityCount('contact'));
+  }
+
+  /**
+   * @return CRM_HRCore_Service_CiviHRStatsGatherer
+   */
+  private function getGatherer() {
+    $siteInformation = $this->prophesize(SiteInformationInterface::class);
+    $siteInformation->getSiteName()->willReturn('foo');
+
+    return new CiviHRStatsGatherer($siteInformation->reveal());
+  }
+
+  /**
+   * @param int $contactID
+   * @param string $startDateString
+   * @param string $endDateString
+   */
+  private function fabricateLeaveRequest(
+    $contactID,
+    $startDateString = 'today',
+    $endDateString = 'tomorrow'
+  ) {
+    LeaveRequestFabricator::fabricateWithoutValidation([
+      'contact_id' => $contactID,
+      'type_id' => $this->absenceTypeID,
+      'from_date' => CRM_Utils_Date::processDate($startDateString),
+      'to_date' => CRM_Utils_Date::processDate($endDateString),
+      'status_id' => 1
+    ], TRUE);
+  }
+
+  /**
+   * @param $contactID
+   */
+  private function setUpLeaveRequest($contactID) {
+    $periodStartDate = CRM_Utils_Date::processDate('-1 year');
+    $periodEndDate = CRM_Utils_Date::processDate('+1 year');
+    $this->absenceTypeID = AbsenceTypeFabricator::fabricate()->id;
+    $absencePeriodID = AbsencePeriodFabricator::fabricate([
+      'start_date' => $periodStartDate,
+      'end_date' => $periodEndDate
+    ])->id;
+    HRJobContractFabricator::fabricate(
+      ['contact_id' => $contactID],
+      ['period_start_date' => $periodStartDate]
+    );
+    LeavePeriodEntitlementFabricator::fabricate([
+      'contact_id' => $contactID,
+      'period_id' => $absencePeriodID,
+      'type_id' => $this->absenceTypeID
+    ]);
+  }
+
+  /**
+   * Gets the number of active contacts now
+   *
+   * @param array $params
+   *
+   * @return int
+   */
+  private function getCurrentContactCount($params = []) {
+    $params['is_deleted'] = 0;
+
+    return (int) civicrm_api3('Contact', 'getcount', $params);
+  }
```

Comment:
I'd prefer not to do this kind of thing in a specific test. Ideally it should be handled by the base test so that we can start each test with a clean slate. I'm not sure if this is the time to write this now though.

Reply:
I'm suggesting to add it to this class because this is the only place where existing records can affect the test results. If we had more classes with tests like that, I would definitely suggest we add it to the base class.

The main reason for doing this is to remove this noise of having to worry about the number of existing entities in a number of tests

Reply:
I'll use the same trait I added in T&A here to truncate tables so.